src/main/java/org/testng/ITestNGMethod.java

```diff
@@ -172,34 +170,15 @@ default boolean hasAfterGroupsConfiguration() {
 
   ITestNGMethod clone();
 
-  /**
-   * @deprecated - This method stands deprecated as of TestNG 7.0.0.
-   * Please use {@link #getRetryAnalyzer(ITestResult)} instead.
-   *
-   * @return The retry analyzer
-   */
-  @Deprecated
-  IRetryAnalyzer getRetryAnalyzer();
-
-  /**
-   * @deprecated - This method stands deprecated as of TestNG 7.0.0.
-   * Please use {@link #setRetryAnalyzerClass(Class)} instead.
-   *
-   * @param retryAnalyzer The retry analyzer
-   */
-  @Deprecated
-  void setRetryAnalyzer(IRetryAnalyzer retryAnalyzer);
-
   default IRetryAnalyzer getRetryAnalyzer(ITestResult result) {
-    return getRetryAnalyzer();
+    return null;

```

Comment:
If you remove the deprecated method, you assume users will have to use the new one. In that case, there is no more default implementation.

Reply:
Fixed